string[] object_names
string instruction
---